tests/shared/test_streamable_http.py

```diff
@@ -1223,3 +1224,64 @@ async def sampling_callback(
                 captured_message_params.messages[0].content.text
                 == "Server needs client sampling"
             )
+
+
+class MockAuthClientProvider:
```

Comment:
I guess we don't need to actually define this class right? Just create a mock.Mock() inside tests? E.g. something like
```python
  client_provider = mock.Mock()
  client_provider.get_headers = AsyncMock(...
```